tests/integration/standard/test_scylla_cloud.py

```diff
@@ -0,0 +1,73 @@
+import os.path
+from unittest import TestCase
+from ccmlib.utils.ssl_utils import generate_ssl_stores
+from ccmlib.utils.sni_proxy import refresh_certs, get_cluster_info, start_sni_proxy, create_cloud_config
+
+from tests.integration import use_cluster
+from cassandra.cluster import Cluster, TwistedConnection
+from cassandra.connection import SniEndPointFactory
+
+
+class ScyllaCloudConfigTests(TestCase):
+    def start_cluster_with_proxy(self):
+        ccm_cluster = self.ccm_cluster
+        generate_ssl_stores(ccm_cluster.get_path())
+        ssl_port = 9142
+        sni_port = 443
+        ccm_cluster.set_configuration_options(dict(
+            client_encryption_options=
+            dict(require_client_auth=True,
+                 truststore=os.path.join(ccm_cluster.get_path(), 'ccm_node.cer'),
+                 certificate=os.path.join(ccm_cluster.get_path(), 'ccm_node.pem'),
+                 keyfile=os.path.join(ccm_cluster.get_path(), 'ccm_node.key'),
+                 enabled=True),
+            native_transport_port_ssl=ssl_port))
+
+        ccm_cluster._update_config()
+
+        ccm_cluster.start(wait_for_binary_proto=True)
+
+        nodes_info = get_cluster_info(ccm_cluster, port=ssl_port)
+        refresh_certs(ccm_cluster, nodes_info)
+
+        docker_id, listen_address, listen_port = \
+            start_sni_proxy(ccm_cluster.get_path(), nodes_info=nodes_info, listen_port=sni_port)
+        ccm_cluster.sni_proxy_docker_id = docker_id
+        ccm_cluster.sni_proxy_listen_port = listen_port
+        ccm_cluster._update_config()
+
+        config_data_yaml, config_path_yaml = create_cloud_config(ccm_cluster.get_path(), listen_port)
+
+        endpoint_factory = SniEndPointFactory(listen_address, port=int(listen_port),
+                                              node_domain="cluster-id.scylla.com")
+
+        return config_data_yaml, config_path_yaml, endpoint_factory
+
+    def test_1_node_cluster(self):
+        self.ccm_cluster = use_cluster("sni_proxy", [1], start=False)
+        config_data_yaml, config_path_yaml, endpoint_factory = self.start_cluster_with_proxy()
+
+        for config in [config_path_yaml, config_data_yaml]:
+            for connection_class in [None, TwistedConnection]:
+                cluster = Cluster(scylla_cloud=config, connection_class=connection_class,
+                                  endpoint_factory=endpoint_factory)
+                with cluster.connect() as session:
+                    res = session.execute("SELECT * FROM system.local")
+                    assert res.all()
+
+                    assert len(cluster.metadata._hosts) == 1
+                    assert len(cluster.metadata._host_id_by_endpoint) == 2
```

Comment:
why it's 2 when cluster has just 1 node?

Reply:
one is `any`, and one is the actual the address with the host_id

Reply:
IMO `any` shouldn't be kept anywhere. It's best to forget about it after initialing a connection. It's endpoint mapping is wrong, because you maye be routed to different host on every reconnection.

Reply:
there are a few of places the control connection uses it's own endpoint to lookup the host.

If we would need to check if we can swap the endpoint on the control connection (at the begin removing of the host cause the control to close)

Reply:
> there are a few of places the control connection uses it's own endpoint to lookup the host.

Where are those places? It shouldn't infer host_id from `any` endpoint, because it's not accurate. 

> If we would need to check if we can swap the endpoint on the control connection (at the begin removing of the host cause the control to close)

I don't understand what you mean, would you mind elaborating more?

Reply:
it's now fixed, and we the extra reference to `any.*` as soon as we can.